### ISSUE TITLE
[FEAT] Implement a redis cache layer

### ISSUE THREAD

**Issue by sellnat77** [hackforla/311-data#543]
### Overview
We have access to a free version of redis on heroku, may as well take advantage of it
See #450 for details

### Action Items
  - [ ] Create redis docker-compose entry in both the actual compose file and the example compose file
  - [ ] Establish cache-key hierarchy (Probably hashed versions of the filter params)
    - ojf84u9r8h44gh9eg -> hashed version of something like (highland park, past 6mo, bulky items)
    - ojf84u9r8h44gh9eg:pins:all
    - ojf84u9r8h44gh9eg:pins:{srnumber}
    - ojf84u9r8h44gh9eg:ttc
    - ojf84u9r8h44gh9eg:freq
  - [ ] Enforce a global TTL for all keys
  - [ ] Ensure connections are short lived
  - [ ] Make redis optional
  - [ ] Use LFU for key recycling

### Resources/Instructions
The caching algorithm will operate similar to this:
User selects filters
front end sends requests to backend
upon receiving request, hash the input parameters
```
Check redis for...
- ojf84u9r8h44gh9eg:pins:all
- ojf84u9r8h44gh9eg:ttc
- ojf84u9r8h44gh9eg:freq
```
If the key is found, retrieve values and return to front end

If nothing is found, continue down normal paths for responses and update the redis keys with the global TTL for 
```
- ojf84u9r8h44gh9eg:pins:all
- ojf84u9r8h44gh9eg:ttc
- ojf84u9r8h44gh9eg:freq
```
then return values to front end

In summary they cache will operate as either a GET then RETURN or a GET, NOTFOUND, SET, RETURN
